modules/services/window-managers/i3-sway/lib/options.nix

```diff
@@ -421,7 +421,13 @@ in {
         };
 
         hideEdgeBorders = mkOption {
-          type = types.enum [ "none" "vertical" "horizontal" "both" "smart" ];
+          type = let
+            i3Options = [ "none" "vertical" "horizontal" "both" "smart" ];
+            swayOptions = i3Options ++ [ "smart_no_gaps" ];
+          in if isI3 then
+            types.enum i3Options
+          else
+            types.enum (swayOptions ++ (map (e: "--i3 ${e}") swayOptions));
```

Comment:
I dont know about the `--i3` flag. What does it do ? couldn't find it at https://github.com/swaywm/sway/blob/master/config.in